src/components/listBox/tests/ListBoxReducers.spec.ts

```diff
@@ -0,0 +1,228 @@
+
+describe('ListBox', () => {
+  describe('ListBox Reducers', () => {
+    const genericAction: IReduxAction<IListBoxPayload> = {
+      type: 'DO_SOMETHING',
+      payload: {
+        id: 'list-box-id',
+      },
+    };
+
+    it('should return the default state if the action is not defined and the state is undefined', () => {
+      const newState: IListBoxState[] = listBoxesReducer(undefined, genericAction);
+
+      expect(newState).toBe(listBoxesInitialState);
+    });
+
+    it('should return the default state if the action is not defined and the state is undefined for one list box', () => {
+      const newState: IListBoxState = listBoxReducer(undefined, genericAction);
+
+      expect(newState).toBe(listBoxInitialState);
+    });
+
+    it('should return the old state when the action is not defined', () => {
+      const oldState: IListBoxState[] = [listBoxInitialState];
+      const newState: IListBoxState[] = listBoxesReducer(oldState, genericAction);
+
+      expect(newState).toBe(oldState);
+    });
+
+    it('should return the old state when the action is not defined for one list box', () => {
+      const oldState: IListBoxState = listBoxInitialState;
+      const newState: IListBoxState = listBoxReducer(oldState, genericAction);
+
+      expect(newState).toBe(oldState);
+    });
+
+    describe('ADD_LIST_BOX', () => {
+      const id = 'added-list-box';
+      const items = [{value: 'a'}, {value: 'b', selected: true}];
+      const selected = _.chain(items).where({selected: true}).pluck('value').value();
+
+      it('should return the old state with one more list box', () => {
+        const oldState: IListBoxState[] = [listBoxInitialState];
+        const newState: IListBoxState[] = listBoxesReducer(oldState, addListBox(id, []));
+
+        expect(newState.length).toBe(oldState.length + 1);
+        expect(newState[1].id).toBe(id);
+      });
+
+      it('should set the selected when there is one in the action', () => {
+        const oldState: IListBoxState[] = [listBoxInitialState];
+        const newState: IListBoxState[] = listBoxesReducer(oldState, addListBox(id, items));
+
+        expect(newState.length).toBe(oldState.length + 1);
+        expect(newState[1].id).toBe(id);
+        expect(newState[1].selected).toEqual(selected);
+      });
+
+      it('should not modify the old state', () => {
+        const oldState: IListBoxState[] = [listBoxInitialState];
+        const oldStateBefore = _.clone(oldState);
+        listBoxesReducer(oldState, addListBox(id, items));
+
+        expect(oldState).toEqual(oldStateBefore);
+      });
+    });
+
+    describe('REMOVE_LIST_BOX', () => {
+      const id = 'exisiting-list-box';
+      let defaultState: IListBoxState[];
+
+      beforeEach(() => {
+        defaultState = [_.extend({}, listBoxInitialState, {id, selected: []}), listBoxInitialState];
+      });
+
+      it('should return the old state with one more list box', () => {
```

Comment:
with one less list box

Reply:
you're missing a test for when the id is not in the state, it should return the previous state (i think it's reducing your coverage too)

Reply:
That copy pasta